eq-author/src/App/page/Design/Validation/AnswerValidation.js

```diff
@@ -151,233 +136,110 @@ export const SidebarValidation = styled(SidebarButton)`
-          })}
-        </React.Fragment>
-      )
-    });
-  };
+  if (validValidationTypes.length === 0) {
+    return;
+  }
 
-  render() {
-    const { answer } = this.props;
+  const validationButtons = [];
+  let pendingErrors = [];
 
-    const validValidationTypes = validations[answer.type] || [];
+  validValidationTypes.forEach(type => {
+    const validation = get(answer, `validation.${type.id}`, {});
+    const errors = get(validation, `validationErrorInfo.errors`, []);
 
-    if (validValidationTypes.length === 0) {
-      return null;
-    }
+    const { enabled, inclusive } = validation;
+    const value = enabled ? type.preview(validation, answer) : null;
 
-    const validationErrors = {
-      dateRange: {
-        duration: [],
-        range: [],
-      },
-      other: [],
+    const onClick = () => {
+      setModalIsOpen(true);
+      setStartingTabId(type.id);
     };
 
-    let validationButtons = [];
-
-    if (answer.type === "DateRange") {
-      const { duration, range } = validValidationTypes.reduce(
-        (accumulator, current) => {
-          const { id } = current;
-          if (id === "minDuration" || id === "maxDuration") {
-            if (!accumulator.duration) {
-              accumulator.duration = [];
-            }
-            accumulator.duration.push(current);
-          }
-          if (id === "earliestDate" || id === "latestDate") {
-            if (!accumulator.range) {
-              accumulator.range = [];
-            }
-            accumulator.range.push(current);
-          }
-          return accumulator;
-        },
-        {}
-      );
-
-      const durationButtons = this.renderValidation(
-        duration,
-        answer,
-        validationErrors
-      );
+    validationButtons.push(
+      <SidebarValidation
+        id={type.id}
+        key={type.id}
+        data-test={`sidebar-button-${kebabCase(type.title)}`}
+        onClick={onClick}
+        hasError={errors.length > 0}
+      >
+        <Title>{titleText(type.id, type.title, enabled, inclusive)}</Title>
+        {enabled && !isNull(value) && <Detail>{value}</Detail>}
+      </SidebarValidation>
+    );
 
-      const rangeButtons = this.renderValidation(
-        range,
-        answer,
-        validationErrors
-      );
+    pendingErrors.push(...errors);
 
-      const durationError = this.renderPropertyError(
-        validationErrors.dateRange.duration.length > 1,
-        DURATION_ERROR_MESSAGE,
-        "duration-error"
-      );
+    const noValError = find(pendingErrors, error =>
+      error.errorCode.includes("ERR_NO_VALUE")
+    );
 
-      const rangeError = this.renderPropertyError(
-        validationErrors.dateRange.range.length > 1,
-        EARLIEST_BEFORE_LATEST_DATE,
-        "range-error"
-      );
+    if (pendingErrors.length > 0) {
+      if ((type.id === "earliestDate" && !noValError) || (type.id === "minDuration" && !noValError) || (type.id === "minValue" && !noValError)) {
```

Comment:
Just a wee suggestion - some redundancy in this boolean expr. Could replace with:

`if(!noValError || type.id === "earliestDate" || type.id === "minDuration" || type.id === "minValue") {`

Reply:
@tbaggaley - that doesn't work as its needed to get the noValError to display immediately after the earliest date (or min duration) button

Reply:
Oops! Yeah, you're right - silly suggestion, sorry! Obvs not the same logic now I look back at it.